Source/Private/common.ps1

```diff
@@ -37,7 +37,10 @@ function _callAPI {
       # that project name to be used in building the URI that would lead to
       # 404 because the URI would not be correct.
       [Alias('IgnoreDefaultProject')]
-      [switch]$NoProject
+      [switch]$NoProject,
+      # this makes sure that you can use your own bearer token on the request if needed
+      # some undocumented internal IPs use their own token based on the PAT
+      [switch]$CustomBearer

```

Comment:
@DarqueWarrior can you please give feedback on this if this ok or if it should be implemented elsewhere? The billing API needs its own token which should not use the PAT. It's based on the PAT but is a JWT token. You get the token with the cmdlet `Get-VSTeamBillingToken`

Reply:
Is there a way to do this without changing this method?

Reply:
Just so I understand. A person that wants to call the billing api calls Get-VSTeamBillingToken before any of the other calls. Then they can all the billing API but need to make sure they don't send the PAT or Bearer token in the call. Is that correct? The JWT is just detected and used when you make the calls that need it?

Reply:
@DarqueWarrior yes this is correct. The PAT is only needed for getting the billing token, which returns a valid JWT token for the billing operation (if the PAT connected to the user has the permissions).

Currently, the only way would be to create a separate method and use that instead of _callAPI But I am not sure if that is good or not.